corehq/form_processor/tests/test_utils.py

```diff
@@ -82,3 +82,18 @@ def test_get_node(self):
         self.assertEqual('remus', get_node(root, '/data/twin[2]/name').text)
         with self.assertRaises(XFormQuestionValueNotFound):
             get_node(root, '/data/has_attribute/@dirty')
+
+
+class TestDateTimeRegexMatch(SimpleTestCase):
+    def test_datetime_match(self):
+        cases = [
+            ('2015-04-03', False),
+            ('2013-03-09T06:30:09.007', False),
+            ('2013-03-09T06:30:09.007+03', True),
+            ('2013-03-09T06:30:09.007-0530', True),
+            ('351602061044374', False),
+            ('2015-01-01T12:00:00.120054Z', True),
+            ('2015-10-01T14:05:45.087434Z', True),
+        ]
```

Comment:
Based on my read of the regex these are also valid values:
```
2020-09-17 12:54+02  # no seconds or milliseconds
2020-09-17 12:54:23+02  # no milliseconds
```

We should confirm that we also want to be converting those formats.